### ISSUE TITLE
Immutable fields

### ISSUE THREAD

**Issue by erikgrinaker** [sanity-io/gradient#543]
@rexxars:
>Proposal: Immutable fields. Fields that can only be set once and never changed (once published).
Can for instance be used to prevent lazy editors from modifying an existing document to mean something different, reusing the same document ID and possibly causing errors.

@erikgrinaker:
>I believe Gradient already has this, in some sense, since e.g. `_type` can not (should not) be changed. This could probably be generalized to arbitrary fields.

**Comment by atombender** [comment]
Easy to do as part of the schema system (which Sanity does not use).

**Comment by simen** [comment]
Yes, but currently immutability is handled through the ImmutablilityChecker which does not integrate with the schema subsystem. Using the schemas would be better in every way.
